### PR TITLE
Encrypt all existing lnm secrets and encrypt new user lnmsecrets

### PR DESCRIPTION
**Pull request by uthpalax**

### CODE REVIEW

File: app/routes/encrypt-all-user-lnmsecrets.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { useLoaderData } from "@remix-run/react";
+import { json } from "@remix-run/node";
+import { prisma } from "~/db.server";
+import { encryptAllSecrets } from "~/models/user.server";
+
+export async function loader() {
```
Comment (dannydeezy) [inline app/routes/encrypt-all-user-lnmsecrets.tsx:6]: when does this get called?

File: app/models/user.server.ts [hunk 2]
```diff
@@ -3,8 +3,51 @@ import bcrypt from "bcryptjs";
...
+      cipher.final(),
+    ]);
+
+    const encryptedLnmSecret = encrypted.toString("hex");
+    const user = await prisma.password.update({
+      where: {
+        userId: data.userId,
+      },
+      data: {
+        lnmSecret: encryptedLnmSecret,
```
Comment (dannydeezy) [inline app/models/user.server.ts]: my concern with overwriting the existing `lnmSecret` is if there's some problem with the encryption/decryption then we'll lose this record. should we just add a new field `encryptedLnmSecret` instead?

Comment (dannydeezy) [reply app/models/user.server.ts]: and then on a second endpoint, remove all existing `lnmSecret`s once we verify decryption is working?

File: app/routes/encrypt-all-user-lnmsecrets.tsx (hunk 1 above)
Comment (uthpalax) [reply app/routes/encrypt-all-user-lnmsecrets.tsx:6]: When we go to that route - /encrypt-all-user-lnmsecrets

I will delete this route after we run it.

File: app/models/user.server.ts (hunk 2 above)
Comment (uthpalax) [reply app/models/user.server.ts]: Yup that is a safer way. 
I did test this locally all is good but we cannot recover is something goes wrong. 
I will use a separate column

File: app/models/user.server.ts [hunk 3]
```diff
@@ -66,15 +66,15 @@ export async function createUser(email: User["email"], password: string) {
...
-  console.log("encrypted", encryptedLnmSecret);
 
   return prisma.user.create({
     data: {
       email,
       password: {
         create: {
           hash: hashedPassword,
-          lnmSecret: encryptedLnmSecret,
+          lnmSecret,
```
Comment (dannydeezy) [inline app/models/user.server.ts:76]: i believe we should remove this `lnmSecret` from here, so that new users are only created w/ `encryptedLnmSecret`

**Review by dannydeezy — APPROVED** [review]
looks good pending the one comment

File: app/models/user.server.ts (hunk 3 above)
Comment (dannydeezy) [reply app/models/user.server.ts:76]: nevermind, good point about verifying its working first before wiping it

**Pull request merged**